pkg/database/workouts_map.go

```diff
@@ -158,37 +156,63 @@ func distanceBetween(p1 gpx.GPXPoint, p2 gpx.GPXPoint) float64 {
 }
 
 func createMapData(gpxContent *gpx.GPX) *MapData {
-	gpxContent.ReduceGpxToSingleTrack()
-
 	if len(gpxContent.Tracks) == 0 {
 		return nil
 	}
 
-	if len(gpxContent.Tracks[0].Segments) == 0 {
-		return nil
+	var totalDistance float64 = 0.0
+	var totalDuration time.Duration
+	var pauseDuration time.Duration
+	var minElevation float64 = 0.0
+	var maxElevation float64 = 0.0
+	var uphill float64 = 0.0
+	var downhill float64 = 0.0
+	var maxSpeed float64 = 0.0
+	var lastSegmentEnd *gpx.GPXPoint
+
+	for _, track := range gpxContent.Tracks {
+		for _, segment := range track.Segments {
+			if len(segment.Points) > 0 {
+				totalDistance += segment.Length3D()
+				totalDuration += time.Duration(segment.Duration()) * time.Second
+				pauseDuration += (time.Duration(segment.MovingData().StoppedTime)) * time.Second
+				minElevation = min(minElevation, segment.ElevationBounds().MinElevation)
+				maxElevation = max(maxElevation, segment.ElevationBounds().MaxElevation)
+				uphill += segment.UphillDownhill().Uphill
+				downhill += segment.UphillDownhill().Downhill
+				maxSpeed = max(maxSpeed, segment.MovingData().MaxSpeed)
+
+				firstPoint := &segment.Points[0]
+				if lastSegmentEnd != nil {
+					// Calculate time between this segment's first point and last segment's last point, and add to pause time
+					pauseDuration += firstPoint.Timestamp.Sub(lastSegmentEnd.Timestamp)
+					pauseDistance := lastSegmentEnd.Point.Distance3D(&firstPoint.Point)
+					totalDistance -= pauseDistance
```

Comment:
I don't think this distance is taken into account when calculating the totalDistance - every "distance" is within a segment, not across segments. Therefore, I think the pauseDistance can be ignored.

Reply:
Yeah, you are right. But then also the pauseDuration should be removed, as that concerns pause that happens outside the segments and should therefore not be counted.